packages/blockchain-extension/extension/commands/addWalletIdentityCommand.ts

```diff
@@ -27,6 +27,20 @@ import { ManagedAnsibleEnvironment } from '../fabric/environments/ManagedAnsible
 import { EnvironmentFactory } from '../fabric/environments/EnvironmentFactory';
 import { LocalEnvironment } from '../fabric/environments/LocalEnvironment';
 
+async function checkIdentityName(identityName: string, identityExists: boolean): Promise<boolean>
+{
+    if (!identityName) {
+        VSCodeBlockchainOutputAdapter.instance().log(LogType.ERROR, `An identity name cannot be empty`);
+        return false;
+    }
+    
+    if (identityExists) {
+        VSCodeBlockchainOutputAdapter.instance().log(LogType.ERROR, `An identity called ${identityName} already exists`);
+        return false;
+    }
+    return true;
+}
+
```

Comment:
As is, this function doesn't need to be async, since everything it does currently happens synchronously.

There are 2 options:
1. Remove the async keyword and the 'await' from the places where it is called;
2. Instead of taking 'identityExists' as an input argument, it can instead take in 'wallet'. This means we would evaluate if the identity exists inside this function, and therefore not require `const identityExists: boolean = await wallet.exists(identityName);` to be called in 3 different locations. It would also mean we won't need to check it until we are sure the identity actually exists, failing sooner if it doesn't. It would also make this function async because 'wallet.exists' is async.

Finally, this function should be at the end of the file, since it is not the most important one. The function `addWalletIdentity` is the one that gives this file it's name, so it should be the first thing we read when we look at this file.